core/src/test/java/ar/com/wolox/wolmo/core/util/ImageProviderTest.kt

```diff
@@ -0,0 +1,186 @@
+package ar.com.wolox.wolmo.core.util
+
+import org.assertj.core.api.Assertions.assertThat
+import android.app.Application
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.graphics.Bitmap
+import android.net.Uri
+import android.provider.MediaStore
+import androidx.fragment.app.Fragment
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.*
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.Implementation
+import org.robolectric.annotation.Implements
+import org.robolectric.annotation.RealObject
+import org.robolectric.shadows.ShadowBitmap
+import java.io.File
+import java.io.IOException
+import java.io.OutputStream
+import java.util.HashMap
+
+@RunWith(RobolectricTestRunner::class)
+@Config(manifest = Config.NONE, shadows = [ImageProviderTest.Companion.WolmoShadowIntent::class, ImageProviderTest.Companion.WolmoShadowBitmap::class])
+class ImageProviderTest {
+
+    companion object {
+        internal var sComponentNameMock: ComponentName? = null
+        internal var sIntentInstance: Intent? = null
+        internal var sCompressionProperties = HashMap<String, String>()
+
+
+        /**
+         * Shadow new instances of [Intent]
+         */
+        @Implements(Intent::class)
+        class WolmoShadowIntent {
+            @RealObject
+            private val realIntent: Intent? = null
+
+            fun __constructor__(action: String) {
```

Comment:
🤔

Reply:
I had to investigate it, because I didn't understand why the variable was named like that. And it's because is a Shadow class, part of Robolectric: http://robolectric.org/extending/